src/librustc_resolve/lib.rs

```diff
@@ -3457,6 +3521,98 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         found_traits
     }
 
+    // Looks up a definitions (and paths) based on a user-supplied predicate
```

Comment:
Nit: Missing period at the end of this sentence -- and I'd say you're filtering with a user-supplied predicate, rather than "based on" -- the definitions are still primarily looked up based on the name, right?